stdenv: add meta.repository attribute to fetchers

AndersonTorres opened the pull request:
Restarting the feature request from lolbinarycat at https://github.com/NixOS/nixpkgs/issues/293838.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Following the hints from 
- https://github.com/NixOS/nixpkgs/pull/294347#discussion_r1544933853 
- https://github.com/NixOS/nixpkgs/pull/294347#discussion_r1544615000

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres committed:
fetchFromGitHub: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres commented:
Why is Nixpkgs manual build failing?

lolbinarycat commented:
just a heads up: i'm going to try to fix meta.repository as soon as i get home (less than an hour), before you go reimplententing the entire thing.

also, if you're going to write an alternative implementation, please title the PR something containing "alternative implementation" otherwise it will get very confusing if i open another pr

AndersonTorres committed:
stdenv: set repository as an empty list by default

AndersonTorres committed:
stdenv: add meta.repository attribute

AndersonTorres commented:
> just a heads up: i'm going to try to fix meta.repository as soon as i get home (less than an hour), before you go reimplententing the entire thing.

Mark me!

> also, if you're going to write an alternative implementation, please title the PR something containing "alternative implementation" otherwise it will get very confusing if i open another pr

As you wish.

AndersonTorres committed:
fetchFromSourceHut: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres committed:
fetchFrom9Front: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres committed:
fetchFromBitbucket: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres committed:
fetchFromGitiles: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres committed:
fetchFromGitLab: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres committed:
fetchFromRepoOrCz: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres committed:
fetchFromSavannah: set meta.repository attribute

So that it can be "inherited" by the caller.

Set it to a list so that it can be "expanded".

AndersonTorres committed:
fetchbzr: set meta.repository attribute

AndersonTorres committed:
fetchcvs: set meta.repository attribute

AndersonTorres committed:
fetchdarcs: set meta.repository attribute

AndersonTorres committed:
fetchfossil: set meta.repository attribute

AndersonTorres committed:
fetchgit: set meta.repository attribute

AndersonTorres committed:
fetchhg: set meta.repository attribute

AndersonTorres committed:
fetchpijul: set meta.repository attribute

AndersonTorres committed:
fetchsvn: set meta.repository attribute

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4123
